library/OrderCloud.Catalyst/DataMovement/ListAllAsync/ListAllExtensions.cs

```diff
@@ -0,0 +1,1077 @@
+			where T : User
+		{
+			return await ListAllHelper.ListAllAsync((page, filter) =>
+			{
+				return resource.ListAsync<T>(supplierID, userGroupID, null, null, SORT, page, MAX_PAGE_SIZE, filters.AndFilter(filter), accessToken);
+			});
+		}
+
+		public static async Task<List<UserGroup>> ListAllAsync(this IUserGroupsResource resource, string buyerID, object filters = null, string accessToken = null)
+		{
+			return await ListAllHelper.ListAllAsync((page, filter) =>
+			{
+				return resource.ListAsync(buyerID, null, null, SORT, page, MAX_PAGE_SIZE, filters.AndFilter(filter), accessToken);
+			});
+		}
+
+		public static async Task<List<T>> ListAllAsync<T>(this IUserGroupsResource resource, string buyerID, object filters = null, string accessToken = null)
+			where T : UserGroup
+		{
+			return await ListAllHelper.ListAllAsync((page, filter) =>
+			{
+				return resource.ListAsync<T>(buyerID, null, null, SORT, page, MAX_PAGE_SIZE, filters.AndFilter(filter), accessToken);
+			});
+		}
+
+		public static async Task<List<UserGroupAssignment>> ListAllUserAssignmentsAsync(this IUserGroupsResource resource, string buyerID, string userGroupID = null, string userID = null, string accessToken = null)
+		{
+			return await ListAllHelper.ListAllAsync((page, filter) =>
+			{
+				return resource.ListUserAssignmentsAsync(buyerID, userGroupID, userID, page, MAX_PAGE_SIZE, accessToken);
+			});
+		}
+
+		public static async Task<List<User>> ListAllAsync(this IUsersResource resource, string buyerID, string userGroupID = null, object filters = null, string accessToken = null)
+		{
+			return await ListAllHelper.ListAllAsync((page, filter) =>
+			{
+				return resource.ListAsync(buyerID, userGroupID, null, null, SORT, page, MAX_PAGE_SIZE, filters.AndFilter(filter), accessToken);
+			});
+		}
+
+		public static async Task<List<T>> ListAllAsync<T>(this IUsersResource resource, string buyerID, string userGroupID = null, object filters = null, string accessToken = null)
+			where T : User
+		{
+			return await ListAllHelper.ListAllAsync((page, filter) =>
+			{
+				return resource.ListAsync<T>(buyerID, userGroupID, null, null, SORT, page, MAX_PAGE_SIZE, filters.AndFilter(filter), accessToken);
+			});
+		}
+
+		public static async Task<List<Webhook>> ListAllAsync(this IWebhooksResource resource, object filters = null, string accessToken = null)
+		{
+			return await ListAllHelper.ListAllAsync((page, filter) =>
+			{
+				return resource.ListAsync(null, null, SORT, page, MAX_PAGE_SIZE, filters.AndFilter(filter), accessToken);
+			});
+		}
+
+		public static async Task<List<XpIndex>> ListAllAsync(this IXpIndicesResource resource, object filters = null, string accessToken = null)
+		{
+			return await ListAllHelper.ListAllAsync((page, filter) =>
+			{
+				return resource.ListAsync(null, null, SORT, page, MAX_PAGE_SIZE, filters.AndFilter(filter), accessToken);
+			});
+		}
+	}
+}
```

Comment:
Is this all handwritten? I think its OK for v1 but we might want to look into generating it from the Open API spec if possible

Reply:
yeah its hand written. AKA potential typo bugs everywhere lol